lib/vfs/src/builder.rs

```diff
@@ -0,0 +1,174 @@
+use crate::{FileSystem, VirtualFile};
+use std::path::{Path, PathBuf};
+use tracing::*;
+use wasmer_wasi_types::types::{__WASI_STDERR_FILENO, __WASI_STDIN_FILENO, __WASI_STDOUT_FILENO};
+
+use super::{NullFile, SpecialFile};
+use super::{TmpFileSystem, ZeroFile};
+
+pub struct RootFileSystemBuilder {
+    default_root_dirs: bool,
+    default_dev_files: bool,
+    add_wasmer_command: bool,
+    stdin: Option<Box<dyn VirtualFile + Send + Sync>>,
+    stdout: Option<Box<dyn VirtualFile + Send + Sync>>,
+    stderr: Option<Box<dyn VirtualFile + Send + Sync>>,
+    tty: Option<Box<dyn VirtualFile + Send + Sync>>,
+}
```

Comment:
Why do we need this?

Reply:
@syrusakbary This is the entrypoint for wasmer.sh. When running bash, some programs assume things like `/bin/wasmer`, `/dev/tty`, etc. to exist (that is why the "special files") exist. The `RootFileSystemBuilder::default().build()` returns a filesystem with the correct files set up.